cli/test/fixtures/static-server.js

```diff
@@ -249,20 +250,26 @@ class Server {
 async function createServers() {
   const servers = [10200, 10503, 10420].map(port => {
     const server = new Server(port);
-    server._server.on('error', e => console.error(e.code, e));
+    server._server.on('error', e => console.error(e.message));
+    if (wasInvokedDirectly) {
+      server._server.on('listening', _ => console.log(`listening on http://localhost:${port}`));
+    }
     return server;
   });
-  await Promise.all(servers.map(s => s.listen(s._port, 'localhost')));
+
+  await Promise.allSettled(servers.map(s => s.listen(s._port, 'localhost'))).then(outcomes => {
+    if (outcomes.every(o => o.status === 'fulfilled')) return;
+    if (outcomes.every(o => o.reason.message.includes('already'))) {
+      return console.warn('😧 Server already up. Continuing…');
+    }
+    console.error(outcomes.map(o => o.reason));
+  });
```

Comment:
For readability, will this work?

Also we should probably throw an error if the servers fail for an unknown reason

```suggestion
  const outcomes = await Promise.allSettled(servers.map(s => s.listen(s._port, 'localhost')));

  if (outcomes.some(o => o.status === 'rejected')) {
    if (outcomes.every(o => o.reason.message.includes('already'))) {
      console.warn('😧 Server already up. Continuing…');
    } else {
      console.error(outcomes.map(o => o.reason));
      throw new Error('One or more servers did not start correctly');
    }
  }

  return servers
```

Reply:
your await flow is definitely better. and fixes the error throwing. (i was ending up with an unhandled rejection previously)

i do still prefer the style of my two every()s, but whatever. definitely not a big deal.

Reply:
@adamraine thanks for writing out the suggestion! :) so clean.